openparticle/utils.py

```diff
@@ -1,16 +1,93 @@
-from openparticle import Fock, ParticleOperator
+from openparticle import *
 import numpy as np
 from typing import List
+import itertools
 
 
-def get_computational_states_fermionic(n_modes):
-    return [
-        Fock([int(j) for j in format(i, "0" + str(n_modes) + "b")][::-1], [], [])
-        for i in range(2**n_modes)
-    ]
+def _fermion_combinations(n):
+    if n is None:
+        return []
+    else:
+        elements = list(range(n + 1))
+        result = []
+        for r in range(1, n + 2):  # Extend range to include the full list
+            result.extend(itertools.combinations(elements, r))
+        return [list(combo) for combo in result]
 
 
-def generate_matrix_from_basis(operator: ParticleOperator, basis: List[Fock]):
+def _give_me_state(nn, Lambda, Omega):
+    # Credit to Kamil Serafin
+    n = nn
```

Comment:
what's the point of calling the variable here `nn` and then redefining it as `n`?